modules/livestore/kafka_client_inmemory.go

```diff
@@ -0,0 +1,329 @@
+package livestore
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/tempo/pkg/ingest"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/plugin/kprom"
+)
+
+// InMemoryMessage represents a message in the in-memory queue
+// Technically there is a fake cluster that listens on a port that is used for other tests.
+// But that has more weight and ties us to actually using Kafka instead of using something similar at a go api level.
+type InMemoryMessage struct {
```

Comment:
this is a mock? testing code? we've placed other testing code here and maybe it belongs there to be useful elsewhere?

https://github.com/grafana/tempo/tree/main/pkg/util/test

again curious what we did in live store/metrics generator given we've not needed this til now.

Reply:
Good place for it. Let me bounce it there.

Reply:
Previously used kfake which simulates kafka over the wire. By moving to a golang api it allows non kafka backed implementations for more flexibility. Especially given the communities preference in single binary to reduce external dependencies.